master/internal/api_trials.go

```diff
@@ -912,6 +912,68 @@ func (a *apiServer) streamMetrics(ctx context.Context,
 	return nil
 }
 
+// Create a TrialSourceInfo, which serves as a link between trials and checkpoints
+// used for tracking purposes for fine tuning and inference.
+func (a *apiServer) ReportTrialSourceInfo(
+	ctx context.Context, req *apiv1.ReportTrialSourceInfoRequest,
+) (*apiv1.ReportTrialSourceInfoResponse, error) {
+	// TODO (Taylor): Handle user auth/rbac
+	resp := &apiv1.ReportTrialSourceInfoResponse{}
+	tsi := req.TrialSourceInfo
+	query := db.Bun().NewInsert().Model(tsi).
+		Value("trial_source_info_type", "?", tsi.TrialSourceInfoType.String()).
+		Returning("trial_id").Returning("checkpoint_uuid")
+	if tsi.GetSourceModelVersionId() == 0 {
+		query.ExcludeColumn("source_model_version_id")
+	}
+	if tsi.GetSourceModelVersionVersion() == 0 {
+		query.ExcludeColumn("source_model_version_version")
+	}
+	_, err := query.Exec(ctx, resp)
+	return resp, err
+}
+
+// Query for all trials that use a given checkpoint and return their metrics.
+func (a *apiServer) GetTrialSourceInfoMetricsByCheckpoint(
```

Comment:
naming nitpick: perhaps call it `GetTrialMetricsBySourceInfoCheckpoint`?.. I feel like that'd better reflect the relationship?
`TrialSourceInfoMetrics` object also feels strange, because it's not metrics on `trial_source_into`, it's the metrics from a trial obtained through a relationship in `trial_source_info`.